src/Castle.Core.Tests/DynamicProxy.Tests/BasicClassProxyTestCase.cs

```diff
@@ -380,6 +380,21 @@ public void ClassProxyShouldCallProtectedDefaultConstructor()
 			Assert.AreEqual("Something", ((ClassWithProtectedDefaultConstructor)proxy2).SomeString);
 		}
 
+		[Test]
+		public void ClassProxyShouldHaveDefaultConstructorWhenBaseClassHasPrivateProtected()
+		{
+			object proxy = generator.CreateClassProxy<ClassWithPrivateProtectedConstructor>();
+			Assert.IsNotNull(Activator.CreateInstance(proxy.GetType()));
+		}
+
+		[Test]
+		public void ClassProxyShouldCallPrivateProtectedDefaultConstructor()
+		{
+			object proxy = generator.CreateClassProxy<ClassWithPrivateProtectedConstructor>();
+			object proxy2 = Activator.CreateInstance(proxy.GetType());
+			Assert.AreEqual("Something", ((ClassWithPrivateProtectedConstructor)proxy2).SomeString);
```

Comment:
Here I'm even less sure I understand why the assertion has to happen on an object created by `Activator`. Why create the first one at all? If you just need the proxy type, you could create that via `generator.ProxyBuilder`... but why not assert on `proxy` directly?